### PR TITLE
Check if MANIFEST.json changes as per #8587

### PR DESCRIPTION
**Pull request by mfeckie**
Adds a script which checks if either of the MANIFEST.json files change in `test/wpt` in response to #8587
- Addresses currently incorrect MANIFEST.json
- Adds checking script

PR for saltfs to add to CI checks to follow (https://github.com/servo/saltfs/pull/163)

https://reviewable.io/reviews/servo/servo/8601

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/8601)

### CODE REVIEW

File: etc/ci/manifest_changed.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+diff=$(git diff -- tests/wpt/**/MANIFEST.json)
```
Comment (Manishearth) [inline etc/ci/manifest_changed.sh]: why the double *?

**Comment by jdm** [comment]
This doesn't actually run the manifest update command, does it?

**Comment by mfeckie** [comment]
@jdm it doesn't run update. I ran the manifest update when running the tests initially to resolve the differences you pointed to the likely came from hand editing of the MANIFEST files. I committed the changes to ensure baseline is correct. Does that make sense? If it's an incorrect approach I can back out that commit.

**Comment by jdm** [comment]
The changes here are correct, but the new script will never report any violations unless we explicitly update the manifest each time it's run, since unlike the check for `Cargo.lock` files, we don't automatically update the manifest during the build.

**Comment by mfeckie** [comment]
@jdm OK, so would auto running the update be an additional step for this PR?  If so, can you point me to the place to start looking :smile:

**Comment by jdm** [comment]
It's a step we need to perform in the script right before checking the result of `git diff`.

**Comment by mfeckie** [comment]
So, run all the tests? Or is there method to just update the manifest?

**Comment by jdm** [comment]
Oh, hrm. That's a good point. @jgraham is there a way to only trigger the manifest update in isolation?

**Comment by eefriedman** [comment]
`./mach test-wpt --manifest-update SKIP_TESTS`?

**Comment by jgraham** [comment]
Something like `./manifest -r -p ../metadata/MANIFEST.json` in `tests/wpt/web-platform-tests` will regenerate the manifest. You would also need to do the same for the mozilla tests passing in `--url-base=/_mozilla/` and the relevant paths under `tests/wpt/mozilla`, and similarly for the CSS tests.

**Comment by mfeckie** [comment]
@jgraham Is there any reason not to use the suggestion of `./mach test-wpt --manifest-update SKIP_TESTS` by @eefriedman ?

**Comment by mfeckie** [comment]
_[etc/ci/manifest_changed.sh, line 2 \[r3\]](https://reviewable.io:443/reviews/servo/servo/8601#-K3UVPNvUICEJIliJkDh-r3-2:-K3b2Z77wPQg_CyRXFzk:-667902993) ([raw file](https://github.com/servo/servo/blob/3f34ee345e7b137847e2e22f9b16d0bae9a454c2/etc/ci/manifest_changed.sh#L2)):_
Will fix, it was because I was using ZSH.

---


_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8601)_

**Comment by jdm** [comment]
Let's run the same for `./mach test-css` too, which is the other MANIFEST.json under tests/wpt/metadata-css.

**Comment by mfeckie** [comment]
@jdm Added.  Running the two commands isn't quick, just want to flag in case it's an issue in terms of test run time.

**Comment by jdm** [comment]
Oh, one last request! We can add the new script to http://mxr.mozilla.org/servo/source/.travis.yml#13 as well for maximum coverage.

**Comment by jdm** [comment]
(Sorry to keep changing the goalposts :)

**Comment by mfeckie** [comment]
@jdm No worries!  As someone coming into the project with limited understanding of how all the parts go together, it's actually been pretty useful!  Extend away :smile:

**Comment by jdm** [comment]
@bors-servo: r+
Hooray!

**Comment by jdm** [comment]
@bors-servo: retry
* #8614

**Pull request merged**